src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -90,13 +104,14 @@ public CSharpType(TypeProvider implementation, string ns, string name, bool isVa
         public IReadOnlyList<CSharpType> Arguments { get; } = Array.Empty<CSharpType>();
         public bool IsFrameworkType => _type != null;
         public Type FrameworkType => _type ?? throw new InvalidOperationException("Not a framework type");
+        public bool IsTypeProvider => _implementation != null;
         public TypeProvider Implementation => _implementation ?? throw new InvalidOperationException($"Not implemented type: '{Namespace}.{Name}'");
```

Comment:
I am not quite sure if we should keep the current names because `Implementation` is kind of not so intuitive
Should we align these two? If so, which should we take?
`IsTypeProvider` and `TypeProvider` or `IsImplemenation` and `Implementation`?

Reply:
We should make the names more straightforward.